e2e/setup-e2e.sh

```
@@ -297,14 +312,26 @@ setup_test_environment() {
     
     cd "$REPO_ROOT"
     
-    # Get the controller endpoint
-    export ENDPOINT=$(helm get values jumpstarter --output json | jq -r '."jumpstarter-controller".grpc.endpoint')
+    # Get the controller endpoint based on deployment method
+    if [ "$METHOD" = "operator" ]; then
+        # For operator deployment, construct the endpoint from the Jumpstarter CR
+        # The operator uses nodeport mode by default with port 8082
+        local BASEDOMAIN=$(kubectl get jumpstarter -n "${JS_NAMESPACE}" jumpstarter -o jsonpath='{.spec.baseDomain}')
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)